services/app-api/src/resolvers/healthPlanPackage/submitHealthPlanPackage.ts

```diff
@@ -357,12 +357,53 @@ export function submitHealthPlanPackageResolver(
             })
         }
 
-        // Since submit can change the form data, we have to save it again.
-        // if the rates were removed, we remove them.
-        let removeRateInfos: RateFormEditableType[] | undefined = undefined
-        if (maybeLocked.rateInfos.length === 0) {
-            // undefined means ignore rates in updaterDraftContractWithRates, empty array means empty them.
-            removeRateInfos = []
+        // If this contract is being submitted as CONTRACT_ONLY but still has associations with rates
+        // we need to prune those rates at submission time to make the submission clean
+        if (
+            contractWithHistory.draftRevision.formData.submissionType ===
+                'CONTRACT_ONLY' &&
+            contractWithHistory.draftRates &&
+            contractWithHistory.draftRates.length > 0
+        ) {
+            const rateUpdates: UpdateDraftContractRatesArgsType = {
+                contractID: contractWithHistory.id,
+                rateUpdates: {
+                    create: [],
+                    update: [],
+                    link: [],
+                    unlink: [],
+                    delete: [],
+                },
+            }
+
+            for (const draftRate of contractWithHistory.draftRates) {
+                if (draftRate.parentContractID !== contractWithHistory.id) {
+                    // this is a linked rate, unlink it
+                    rateUpdates.rateUpdates.unlink.push({
+                        rateID: draftRate.id,
+                    })
+                } else if (draftRate.revisions.length === 0) {
+                    // this is a child draft rate, delete it
+                    rateUpdates.rateUpdates.delete.push({
+                        rateID: draftRate.id,
+                    })
+                } else {
+                    // this is a previously submitted child-rate. I'm not sure what to do with it
+                    console.warn(
+                        'Attempting to remove a previously submitted child rate in submit',
+                        contractWithHistory.id
+                    )
+                    throw new Error('no nonono bad child rate submit')
```

Comment:
This is not supposed to be an error? This is supposed to just be unlinked from its parent but not deleted from db. If a user does this valid behavior in the app and we error that means we are preventing valid actions in the app I think??

Reply:
No, this supports that. The submitRate function calls this passing an undefined contractID, and it sets all the package submission stuff correctly. My goal was for all that bookkeeping to happen in one place, but submitRate works great with it on its own. And yes, I can move this to its own file to make that more clear.

Reply:
This check is only running if someone has switched a contract to CONTRACT_ONLY from one where they had rates when it was CONTRACT_AND_RATES, but I think you are right that we should treat this the same as unlink.